primitives/traits/src/lib.rs

```diff
@@ -1052,3 +1052,23 @@ pub trait UpdateStoragePrice {
         total_capacity: Self::StorageDataUnit,
     ) -> Self::Price;
 }
+
+/// A trait to calculate the percentage of charge funds by a Provider that should go to the treasury.
+///
+/// This is used by the Payment Streams pallet, which requires some type to implement this trait,
+/// and uses such implementation to calculate the percentage of charged funds by a Provider that should go to the treasury.
+pub trait TreasuryCutCalculator {
+    /// The numerical type which represents the balance type of the runtime.
+    type Balance: NumericalParam;
+    /// Type of the unit provided by Providers
+    type ProvidedUnit: NumericalParam + Into<u64>;
```

Comment:
Yep, the thing is we have that limitation in all other interfaces that use the `StorageDataUnit` or `ProvidedUnit` type. In the future we should explore if it's worth to refactor our logic in a way to not require that limitation

Reply:
Yeah, it's not needed in the interface itself because all implementations I've done also have the `Into<u64>` limitation, that's why it compiles even after removing it from the trait. Considering it gives the trait more flexibility I'll remove it from its declaration